projects/09/Snake/Game.jack

```diff
@@ -0,0 +1,153 @@
+class Game {
+	field int screenWidth, screenHeight, topLimit, bottomLimit, rightLimit, leftLimit, borderSize;
+	field int score;
+	field Snake snake;
+	field Random rand;
+	field boolean foodExists;
+	field Food food;
+
+    /** Constructs a new Game object. */
+    constructor Game new(int screenWidthIn, int screenHeightIn, int borderSizeIn) {
+		let borderSize = borderSizeIn;
+		let screenWidth = screenWidthIn;
+		let screenHeight = screenHeightIn;
+        let topLimit = borderSize;
+        let bottomLimit = screenHeightIn - borderSize;
+        let rightLimit = screenWidthIn - borderSize;
+        let leftLimit = borderSize;
+		let score = 0;
+		do createSnake();
+        return this;
+    }
+
+    /** Deallocates the object's memory. */
+    method void dispose() {
+		do snake.dispose();
+		do rand.dispose();
+		do food.dispose();
+        do Memory.deAlloc(this);
+        return;
+    }
+	
+	method void createSnake() {
+		let snake = Snake.new(15);
```

Comment:
you can define the constant as self describing static fields

Reply:
I solved it by extracting the block size and starting position to be CTOR arguments